### PR TITLE
Modernize to latest versions supported by Java 8

### PR DESCRIPTION
**Pull request by sghill-rewrite**
Hi @oleg-nenashev!

This PR aims to modernize tooling and move this plugin closer to the [recommended Jenkins baseline version](https://www.jenkins.io/doc/developer/plugin-development/choosing-jenkins-baseline/) without making the change too big. If this goes well, I can follow up with a subsequent PR to modernize further.

Use this link to re-run the recipe: https://app.moderne.io/recipes/org.openrewrite.jenkins.ModernizePluginForJava8

### Testing done

Ran `mvn clean verify`. The new tooling picked up an existing SpotBugs issue, but after seeing the comments I decided to suppress this instance because it's clearly intentional.

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

Fixes #62.

### CODE REVIEW

**Comment by sghill** [comment]
@NotMyFault I saw you merged a recent PR in this repo. Would you be able to look at this modernization PR or recommend someone who could?

**Comment by NotMyFault** [comment]
> NotMyFault I saw you merged a recent PR in this repo. Would you be able to look at this modernization PR or recommend someone who could?

The PR looks good, but I'm not a maintainer of this plugin, just merged some infra chores. 
The plugin is up for adoption, currently.

File: src/main/java/com/cloudbees/jenkins/plugins/customtools/CustomToolInstallWrapper.java [hunk 1]
```diff
@@ -221,6 +221,7 @@ public Launcher decorateLauncher(AbstractBuild build, final Launcher launcher,
 
         return new Launcher.DecoratedLauncher(launcher) {
             @Override
+            @SuppressFBWarnings("DCN_NULLPOINTER_EXCEPTION")
```
Comment (oleg-nenashev) [inline src/main/java/com/cloudbees/jenkins/plugins/customtools/CustomToolInstallWrapper.java:224]: Okay, justification is below in the comments....

**Comment by sghill** [comment]
Thank you @oleg-nenashev!

**Pull request merged**